python/TestHarness/schedulers/Job.py

```diff
@@ -53,6 +54,26 @@ def getDAG(self):
         """ Return the DAG object """
         return self.__dag
 
+    def getOriginalDAG(self):
+        """
+        Retreive the DAG object from the state it was when setOriginalDAG was called or the current
+        state it is in now, if setOriginalDAG was never called.
+        """
+
+        return self.setOriginalDAG()
+
+    def setOriginalDAG(self):
+        """
+        Create a soft clone of the working DAG for what ever state it is currently in. This method
+        should only be called once, and once the working DAG is properly set up.
+
+        This is to protect the DAG from further tampering.
+        """
+
+        if self.__dag_clone == None:
+            self.__dag_clone = self.__dag.clone()
+        return self.__dag_clone
+
```

Comment:
_really_ wish I can make this somehow 'automatic' when the class gets instanced. But the desired DAG does not get created until later.

Is there a way to do what I am asking with out an external call to setOriginalDAG ?